app/models/manageiq/providers/vmware/infra_manager/provision/configuration.rb

```diff
@@ -5,11 +5,28 @@ module ManageIQ::Providers::Vmware::InfraManager::Provision::Configuration
   include_concern 'Network'
   include_concern 'Disk'
 
-  def reconfigure_hardware_on_destination?
+  def reconfigure_container_on_destination?
     # Do we need to perform a post-clone hardware reconfigure on the new VM?
     [:cpu_limit, :memory_limit, :cpu_reserve, :memory_reserve].any? do |k|
       return false unless options.key?(k)
       destination.send(k) != options[k]
     end
   end
+
+  def reconfigure_disk_on_destination?(destination)
+    default_size = destination.hardware.disks.find_by(:device_type => "disk").size / (1024**3)
+    get_option(:disk_size) > default_size
+  end
+
+  def reconfigure_hardware(destination)
+    config_spec = VimHash.new("VirtualMachineConfigSpec") do |vmcs|
+      set_cpu_and_memory_allocation(vmcs) if reconfigure_container_on_destination?
+      set_disk_allocation(destination, vmcs) if reconfigure_disk_on_destination?(destination)
+    end
+    unless config_spec.empty?
+      _log.info("Calling VM reconfiguration")
+      dump_obj(config_spec, "#{_log.prefix} Post-create Config spec: ", $log, :info)
+      vm.spec_reconfigure(config_spec)
+    end
```

Comment:
Minor style thing but we typically prefer to

```suggestion

    return if config_spec.empty?
```

Over adding the rest of the method in a conditional

Reply:
fixed